specs/dynamicdependencies/DynamicDependencies.md

```diff
@@ -0,0 +1,1409 @@
+namespace Microsoft.ApplicationModel.DynamicDependencies
+{
+/// CPU architectures to optionally filter available packages against a package dependency.
+/// These generally correspond to processor architecture types supported by MSIX.
+/// @see Windows.System.ProcessorArchitecture
+enum PackageDependencyProcessorArchitectures
+{
+    None       = 0,
+    Neutral    = 0x00000001,
+    X86        = 0x00000002,
+    X64        = 0x00000004,
+    Arm        = 0x00000008,
+    Arm64      = 0x00000010,
+    X86OnArm64 = 0x00000020,
+};
+
+enum PackageDependencyLifetimeArtifactKind
+{
+    /// The current process is the lifetime artifact. The package dependency
+    /// is implicitly unpinned when the process terminates.
+    Process,
+
+    /// The lifetime artifact is an absolute filename or path.
+    /// The package dependency is implicitly unpinned when this is deleted.
+    FilePath,
+
+    /// The lifetime artifact is a registry key in the format
+    /// 'root\\subkey' where root is one of the following: HKLM, HKCU, HKCR, HKU.
+    /// The package dependency is implicitly unpinned when this is deleted.
+    RegistryKey,
+};
+
+/// Options when 'pinning' a package dependency
+runtimeclass PinPackageDependencyOptions
+{
+    PinPackageDependencyOptions();
+
+    /// Optional filtering by cpu architecture(s)
+    PackageDependencyProcessorArchitectures Architectures;
+
+    /// Do not verify at least 1 matching package exists when pinning a package dependency
+    Boolean VerifyDependencyResolution = true;
+
+    /// The kind of lifetime artifact for this package dependency.
+    PackageDependencyLifetimeArtifactKind LifetimeArtifactKind;
+
+    /// The lifetime artifact when pinning a package dependency. The value depends on the LifetimeArtifactKind value.
+    String LifetimeArtifact;
+}
+
+/// Options when adding a package dependency
+runtimeclass AddPackageDependencyOptions
+{
+    AddPackageDependencyOptions();
+
+    /// The rank when adding the package dependency to a a package graph.
+    /// @note A package graph is sorted in ascending order from -infinity...0...+infinity
+    /// @note The default value is zero
+    Int32 Rank;
+
+    /// If a package dependency resolves to a package with dependencies and this option is true
+    /// the dependencies are ignored; only the package is added to the package graph. By default
+    /// a resolved package and its dependencies are added to the package graph.
+    Boolean OnlyUseFirstPackageFamily;
+
+    /// If a package is added to a package graph with a package of the same rank (aka a collision on rank)
+    /// and this option is true the package is prepended to the set of packages of the same rank.
+    /// By default packages are appended to the set of packages with the same rank.
+    Boolean PrependIfRankCollision;
+}
+
+/// TBD
+runtimeclass PackageDependency
+{
+    /// Create an intstance of the package dependency identified by id.
+    PackageDependency(String id);
```

Comment:
I think this is really a `GetFromId` or `FromId`, since it's recovering an existing object, not creating a new one. People would be drawn to this constructor (because constructor) and and go nuts trying to figure out how to create an Id.

Reply:
RECOMMEND: Switch to `static PackageDependency GetFromId(String id)`